driver-core/src/test/functional/com/mongodb/internal/connection/DefaultConnectionPoolTest.java

```diff
@@ -316,7 +317,6 @@ public void concurrentUsage(final int minSize, final int maxSize, final boolean
                 ConnectionPoolSettings.builder()
                     .minSize(minSize)
                     .maxSize(maxSize)
-                    .maxWaitTime(TEST_WAIT_TIMEOUT_MILLIS, MILLISECONDS)
```

Comment:
During my review, I noticed that in a previous PR, `maxTimeAwait()` was removed from the `concurrentUsage()` test. I understand the rationale, assuming that we no longer use the maxTimeAwait parameter from ConnectionPool in `DefaultConnectionPool` in favor of `TimeoutContext`.

However, upon further examination, I found that the sync method (as seen [here](https://github.com/mongodb/mongo-java-driver/blob/272a7c41e75c394d753fa8901998e5b0999a7646/driver-core/src/main/com/mongodb/internal/connection/DefaultConnectionPool.java#L190)) uses `maxAwaitTime` from `TimeoutContext`, while `getAsync` from `ConnectionPoolSettings`.

As a result, the `maxAwaitTime` limitation in the test applies only for the sync version of execution, but we test both sync and async. Should we consider leaving the `maxAwaitTime` parameter in the test until `TimeoutContext` is used in `getAsync`? Otherwise, the test may not be complete.

Reply:
Yeah, I missed this one because it's a Slow test that didn't run in the patch build.  Fixed in the followup PR.